Fix error on empty LUABIN_VERSION_STRING

As @sirblackheart reported in #744 cmake fails when lua is not installed with:

```
CMake Error at cmake/Modules/FindLua.cmake:117 (string):
  string sub-command SUBSTRING requires four arguments.
Call Stack (most recent call first):
  cmake/Modules/FindLua.cmake:196 (verify_lua_executable_version)
  src/plugins/lua/CMakeLists.txt:2 (find_package)
```

Affected are http://community.markus-raab.org:8080/job/elektra-gcc-configure-debian-wheezy/lastFailedBuild/console http://community.markus-raab.org:8080/job/elektra-gcc47-all/lastFailedBuild/console http://community.markus-raab.org:8080/job/elektra-git-buildpackage-wheezy/lastFailedBuild/console http://community.markus-raab.org:8080/job/elektra-icc/lastFailedBuild/console

So a fix is highly appreciated.

--- a/cmake/Modules/FindLua.cmake
+++ b/cmake/Modules/FindLua.cmake
@@ -97,34 +97,39 @@ endfunction(set_lua_version_vars)
 set_lua_version_vars()
 
 function (verify_lua_executable_version)
-	find_program (LUA_EXECUTABLE
-		NAMES
-		"lua-${_LUA_VERSION_MAJOR}.${_LUA_VERSION_MINOR}"
-		"lua${_LUA_VERSION_MAJOR}.${_LUA_VERSION_MINOR}"
-		"lua${_LUA_VERSION_MAJOR}${_LUA_VERSION_MINOR}"
-		"lua"
-		PATH
-	)
-	
-	execute_process (COMMAND ${LUA_EXECUTABLE} "-v" 
-		OUTPUT_VARIABLE LUABIN_VERSION_STRING 
-		ERROR_VARIABLE LUABIN_VERSION_STRING
-		OUTPUT_STRIP_TRAILING_WHITESPACE
-		ERROR_STRIP_TRAILING_WHITESPACE
-	)
-	
-	# Compare only MAJOR.MINOR
-	string (SUBSTRING ${LUABIN_VERSION_STRING} "4" "3" LUABIN_VERSION_STRING)
-	string (COMPARE EQUAL ${LUABIN_VERSION_STRING} "${_LUA_VERSION_MAJOR}.${_LUA_VERSION_MINOR}" VERSION_MATCHES)
-	
-	if (NOT VERSION_MATCHES)
-		message ("Warning: Lua executable does not match lua library version")
-		message ("Lua library: ${_LUA_VERSION_MAJOR}.${_LUA_VERSION_MINOR}")
-		message ("Lua executable: ${LUABIN_VERSION_STRING}")
-		set (LUA_EXECUTABLE_VERSION_MATCHED FALSE PARENT_SCOPE)
-	else ()
-		set (LUA_EXECUTABLE_VERSION_MATCHED TRUE PARENT_SCOPE)
-	endif()
+    set (LUA_EXECUTABLE_VERSION_MATCHED FALSE PARENT_SCOPE)
+    
+    find_program (LUA_EXECUTABLE
+        NAMES
+        "lua-${_LUA_VERSION_MAJOR}.${_LUA_VERSION_MINOR}"
+        "lua${_LUA_VERSION_MAJOR}.${_LUA_VERSION_MINOR}"
+        "lua${_LUA_VERSION_MAJOR}${_LUA_VERSION_MINOR}"
+        "lua"
+        PATH
+    )
+
+    execute_process (COMMAND ${LUA_EXECUTABLE} "-v"
+        OUTPUT_VARIABLE LUABIN_VERSION_STRING
+        ERROR_VARIABLE LUABIN_VERSION_STRING
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+        ERROR_STRIP_TRAILING_WHITESPACE
+    )
+
+    if (LUA_EXECUTABLE AND (NOT LUABIN_VERSION_STRING STREQUAL ""))
+        # Compare only MAJOR.MINOR
+        string (SUBSTRING ${LUABIN_VERSION_STRING} "4" "3" LUABIN_VERSION_STRING)
+        string (COMPARE EQUAL ${LUABIN_VERSION_STRING} "${_LUA_VERSION_MAJOR}.${_LUA_VERSION_MINOR}" VERSION_MATCHES)
+
+        if (NOT VERSION_MATCHES)
+            message (WARNING "Lua executable does not match lua library version")
+            message ("Lua library: ${_LUA_VERSION_MAJOR}.${_LUA_VERSION_MINOR}")
+            message ("Lua executable: ${LUABIN_VERSION_STRING}")
+        else ()
+            set (LUA_EXECUTABLE_VERSION_MATCHED TRUE PARENT_SCOPE)
+        endif(NOT VERSION_MATCHES)
+    else ()
+        message (WARNING "Lua executable not found")
+    endif(LUA_EXECUTABLE AND (NOT LUABIN_VERSION_STRING STREQUAL ""))
 endfunction (verify_lua_executable_version)
 
 find_path(LUA_INCLUDE_DIR lua.h

